main.cpp

```diff
@@ -178,17 +178,17 @@ struct Shoe
 
 void Shoe::putOn(std::string firstShoe) 
 {
- 
+    firstShoe = "R";
 }
 
 void Shoe::takeOff(std::string firstShoe) 
 {
- 
+    firstShoe = "R";
 }
 
 void Shoe::throwAway(bool donate) 
 {
- 
+    donate = true;
 }
```

Comment:
are you aware that you are overwriting the value of the variable you're passing **INTO** the function, thus defeating the purpose of even passing anything into the function...
- [ ] revise